erpnext/stock/doctype/quality_inspection/quality_inspection.js

```diff
@@ -33,19 +32,25 @@ frappe.ui.form.on("Quality Inspection", {
 			});
 		}
 	},
-	onload: (frm) => {
-		if (frm.doc.item_code) {
-			frm.trigger("check_compliance_item");
-		}
-	},
 	check_compliance_item: function (frm) {
 		frappe.db.get_value("Compliance Item", { "item_code": frm.doc.item_code }, "item_code")
 			.then(item => {
 				frm.toggle_reqd('batch_no', !!item.message);
```

Comment:
The other fields seem fine to me, but shouldn't batch no be mandatory on selection of item?